charts/consul/templates/server-statefulset.yaml

```diff
@@ -8,6 +8,9 @@
 {{- if (and .Values.global.gossipEncryption.secretName (not .Values.global.gossipEncryption.secretKey)) }}{{fail "gossipEncryption.secretKey and secretName must both be specified." }}{{ end -}}
 {{- if (and (not .Values.global.gossipEncryption.secretName) .Values.global.gossipEncryption.secretKey) }}{{fail "gossipEncryption.secretKey and secretName must both be specified." }}{{ end -}}
 {{- if (and .Values.global.secretsBackend.vault.enabled (not .Values.global.secretsBackend.vault.consulServerRole)) }}{{ fail "global.secretsBackend.vault.consulServerRole must be provided if global.secretsBackend.vault.enabled=true." }}{{ end -}}
+{{- if (and .Values.server.serverCert.secretName (not .Values.global.tls.caCert.secretName) (not .Values.global.secretsBackend.vault.enabled)) }}{{ fail "If server.serverCert.secretName is provided, global.tls.caCert must also be provided" }}{{ end }}
```

Comment:
isn't that still true when vault is enabled, meaning you need to provide global.tls.caCert.secretName when `Values.server.serverCert.secretName` is also set?

Reply:
good catch, this was leftover :) thanks!